src/documentation/documentationengine/documentationviewer.cpp

```diff
@@ -0,0 +1,227 @@
+    mPrevPageAction = pPageMenu->addAction("&Prev", mWebView, &QWebEngineView::back);
+    mRefreshPageAction = pPageMenu->addAction("&Refresh", mWebView, &QWebEngineView::reload);
+    mStopLoadAction = pPageMenu->addAction("&Stop", mWebView, &QWebEngineView::stop);
+
+    mNextPageAction->setIcon(QIcon::fromTheme("go-next"));
+    mPrevPageAction->setIcon(QIcon::fromTheme("go-previous"));
+    mRefreshPageAction->setIcon(QIcon::fromTheme("view-refresh"));
+    mStopLoadAction->setIcon(QIcon::fromTheme("process-stop"));
+
+
+    QToolBar *pToolBar = addToolBar("Main Tool Bar");;
+    pToolBar->addAction(mPrevPageAction);
+    pToolBar->addAction(mNextPageAction);
+    pToolBar->addAction(mRefreshPageAction);
+    pToolBar->addAction(mStopLoadAction);
+    pToolBar->addWidget(mUrlEdit);
+
+    menuBar()->addMenu(pViewMenu);
+    menuBar()->addMenu(pPageMenu);
+
+    connect(mWebView, &QWebEngineView::urlChanged, this, &DocumentationViewer::updateUrlBar);
+    connect(mUrlEdit, &QLineEdit::returnPressed, this, &DocumentationViewer::urlRequested);
+    connect(mWebView, &QWebEngineView::titleChanged, this, &DocumentationViewer::titleChange);
+
+
+    pToolBar->addWidget(mStackedWidget);
+    mStackedWidget->addWidget(mUrlEdit);
+    mStackedWidget->addWidget(mProgressBar);
+
+    mStackedWidget->setCurrentWidget(mUrlEdit);
+    connect(mWebView, &QWebEngineView::loadStarted, this, &DocumentationViewer::loadStarted);
+    connect(mWebView, &QWebEngineView::loadFinished, this, &DocumentationViewer::loadFinished);
+    connect(mWebView, &QWebEngineView::loadProgress, mProgressBar, &QProgressBar::setValue);
+    setWindowIcon(mWebView->icon());
+
+    mCurrentPage = mWebView->page();
+
+    addToolBar(Qt::TopToolBarArea, pToolBar);
+
+    connect(mCurrentPage, &QWebEnginePage::linkHovered, this, &DocumentationViewer::linkHovered);
+
+}
+
+QWebEngineView *DocumentationViewer::webView() const
+{
+    return mWebView;
+}
+
+void DocumentationViewer::loadReferenceDocumentation()
+{
+    bool isPairStormExist;
+    QDir dir = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+
+    isPairStormExist = dir.cd("PairStorm");
+
+    if(mConnectionManager->hasConnection())
+    {
+        mWebView->load(QUrl("https://en.cppreference.com/w/"));
+    }
+    else
+    {
+        dir.cd("reference");
+        dir.cd("en");
+        QString indexPath{dir.path() + "/" + "index.html"};
+        mWebView->load(QUrl::fromLocalFile(indexPath));
+    }
+}
+
+void DocumentationViewer::loadReferenceDocumentation(const QString &keyword)
+{
+    bool isPairStormExist;
+    QDir dir = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+
+    isPairStormExist = dir.cd("PairStorm");
+
+    if(!isPairStormExist)
+    {
+        if(mConnectionManager->hasConnection())
+        {
+            mWebView->load(QUrl("https://en.cppreference.com/w/"));
+        }
+    }
```

Comment:
You use same urls several times.
It might have sense to replace them with constant